### PR TITLE
Added code to convert SQuAD data into a sequence tagging task

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: src/main/scala/org/allenai/deep_qa/parse/Parser.scala [hunk 1]
```diff
@@ -86,6 +87,12 @@ class StanfordParser extends Parser {
     parsed
```
Comment (DeNeutoy) [inline src/main/scala/org/allenai/deep_qa/parse/Parser.scala:87]: Something weird is happening here to make the `Parser` tests fail, but I can't see what it is - the types match up ...

File: src/main/scala/org/allenai/deep_qa/data/Instance.scala [hunk 2]
```diff
@@ -152,3 +152,32 @@ case class McQuestionAnswerInstance(
     Seq(Seq(s"$passage\t$question\t$answerString" + labelString))
   }
 }
+
+trait BaseTaggingInstance extends Instance
```
Comment (DeNeutoy) [inline src/main/scala/org/allenai/deep_qa/data/Instance.scala:156]: Why is this required?

File: src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    for (candidateSpan <- sortedSpans) {
+      val candidateSpanStart = candidateSpan._1
+      val candidateSpanEnd = candidateSpanStart + candidateSpan._2.size
+      val overlaps = keptSpans.map { case (spanStart, spanText) => {
+        val spanEnd = spanStart + spanText.size
+        val beginOverlaps = candidateSpanStart >= spanStart && candidateSpanStart <= spanEnd
+        val endOverlaps = candidateSpanEnd >= spanStart && candidateSpanStart <= spanEnd
+        beginOverlaps || endOverlaps
+      }}
+      if (!overlaps.foldLeft(false)(_ || _)) {
```
Comment (DeNeutoy) [inline src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala]: I know this is very idiomatic Scala but I think it would be easier to read if this was just `!overlaps.exists(_ == true)`.

**Review by DeNeutoy — APPROVED** [review]

File: src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala (hunk 3 above)
Comment (DeNeutoy) [reply src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala]: Ok I just discussed this with Marc because I was interested what he thought and he convinced me that your way is better, because the logic is explicit, whereas with `exists` it is implicitly `and`. I retract my comment 😄

File: src/main/scala/org/allenai/deep_qa/data/Instance.scala (hunk 2 above)
Comment (matt-gardner) [reply src/main/scala/org/allenai/deep_qa/data/Instance.scala:156]: It's not required, it just gives us a more specific type to return for a sequence tagging dataset reader.  So, e.g., if you want to enumerate all possibilities with a `match ... case` statement, you have many fewer to worry about.

File: src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala (hunk 3 above)
Comment (matt-gardner) [reply src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala]: Actually, I like the `overlaps.exists(_ == true)` better =).  I was trying to figure out how to do that with a `reduce`, but it didn't work, so switched to `foldLeft`.  If I had remembered `exists` when I was writing this, I would have used it.

Comment (matt-gardner) [reply src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala]: ~~Note that it's `overlaps.exists(_ == true)` and not `!overlaps.exists(_ == true)`, though~~ (duh... I should not make these comments early in the morning...).  And I don't think there's anything implicit in using `exists` - the contract of the method is an `or`.

File: src/main/scala/org/allenai/deep_qa/data/Instance.scala (hunk 2 above)
Comment (DeNeutoy) [reply src/main/scala/org/allenai/deep_qa/data/Instance.scala:156]: Ah ok, got it.
